src/JWT/Serializers/SystemTextJsonSerializer.cs

```diff
@@ -0,0 +1,58 @@
+﻿#if SYSTEMTEXTJSON
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace JWT.Serializers
+{
+    public sealed class SystemTextJsonSerializer : IJsonSerializer
+    {
+        public string Serialize(object obj) =>
+            JsonSerializer.Serialize(obj);
+
+        public T Deserialize<T>(string json)
+        {
+            var data = JsonSerializer.Deserialize<T>(json);
+
+            // when deserializing a Dictionary<string, object>
+            // System.Text.Json create JsonElement objects for every value of the dictionary
+            // but application will expect to have native basic types (not a JsonElement class)
+            if (!(data is Dictionary<string, object> odata))
```

Comment:
Do you know how it'll work if the type is IDictionary or IReadOnlyDictionary?

Reply:
I suppose it work out of the box with IDictionary (the concrete type returned by JsonSerializer.Deserialize<**IDictionary**<string, object>> is **Dictionary**<string, object>)
As I understand, the same happens with IReadOnlyDictionary, the concrete type deserialized is always a **Dictionary**<string, object>.

Reply:
Great! Thanks for the clarification. I'll add tests once we decide how to handle this in general.

Reply:
Hmm, I see. But direct cast to T won't work likely. Would `(T)(object)ndata` work here little better? Maybe the effect is same but more readable.

Reply:
Yes, it works, but the effect is not exactly the same ... if the double conversion don't work, null will be returned.

`return (T)(object)ndata ?? data` is the equivalent statement but I don't know if it is more readable ... I don't like double casting ... 

`if (ndata is T obj) return obj; else return data;` is another, more human readable, option ...